internal/clients/clients.go

```diff
@@ -295,15 +299,18 @@ func (c *Cache) Get(cr auth.Credentials, o ...GetOption) (client.Client, error)
 	// is possible to 'reset' (i.e. extend) a ticker.
 	expiration := &tickerExpiration{t: time.NewTicker(c.expiry)}
 	newExpiry := time.Now().Add(c.expiry)
-	ctx, cancel := context.WithCancel(context.Background())
-	sn = &session{client: wc, cancel: cancel, expiry: c.expiry, expiration: expiration, log: log}
+	ctx, cancel := context.WithCancel(c.ctx)
```

Comment:
use `Cache`'s context for constructing the client instead of creating a new context.